### ISSUE TITLE
20.0.4 -> 20.0.5 resets shared folder URL's

### ISSUE THREAD

**Issue by SimplyCorbett** [nextcloud/server#25169]
If you need more details let me know.

This folder also has 10,000 files so I'm not sure if that's why this happened.

### Steps to reproduce
1. Upgrade 20.0.4 to 20.0.5
2. URL's are "reset". Well, to be more accurate it says there are no files available in the folder.
3. Create a new shared link for the folder, works.

**Operating system:**
Debian 10

**Web server:**
Nginx mainline

**Database:**
Postgresql 13
**PHP version:**
PHP 7.4 FPM
**Nextcloud version:** (see Nextcloud admin page)
20.0.4 -> 20.0.5
**Updated from an older Nextcloud/ownCloud or fresh install:**
Upgraded
**Where did you install Nextcloud from:**
nextclouds main website.

**Comment by derBobby** [comment]
Whitespaces in the URL? Might be same issue like #25164 #25165

**Comment by Luticus** [comment]
I thought about creating a separate bug for this, but decided there's a good chance all this sharing stuff is related, but 20.0.5 broke sharing entirely for me. When someone goes to use the share icon it pops out the side panel and just spins, never loading. I downgraded to 20.0.4 and it worked perfectly again. I think I'm using php 7.3fpm on Debin with a mariadb 10.3 database. Also running Apache 2.4.38. I'm using the LDAP and LDAP write plugins as well.

**Comment by llucax** [comment]
+1 on having to rollback an upgrade from 20.0.4 to 20.0.5 because several apps didn't work anymore, including Deck and Calendar.

**Comment by pizzaboy75** [comment]
Has this Issue been resolved in 20.0.6?

**Comment by Luticus** [comment]
I deployed 20.0.6 and it looks like the sharing interface is working. I had some issues with the news app where I had to pull the app out of my 20.0.4 nextcloud folder and move it to 20.0.6/apps because when I went into the store to reinstalled it, it gave me a failed to extract error. Had to do the same with the jsxc app because I couldn't find it in the app store for some reason. Luckily they both easily enable via the app store once moved in place. There were a hand full of other apps I needed to re-enable like rainloop, but other than that everything *seems* fine.
